src/com/ms/silverking/thread/lwt/LWTQueueImpersonator.java

```diff
@@ -15,154 +15,153 @@
-    public int size() {
-        return 0;
-    }
-
-    @Override
-    public Object[] toArray() {
-        throw new UnsupportedOperationException("Method not implemented");
-    }
-
-    @Override
-    public <T> T[] toArray(T[] a) {
-        throw new UnsupportedOperationException("Method not implemented");
-    }
+  private final BaseWorker<T> worker;
+  private final int priority;
+
+  public LWTQueueImpersonator(BaseWorker<T> worker, int priority) {
+    this.worker = worker;
+    this.priority = priority;
+  }
+
+  public LWTQueueImpersonator(BaseWorker<T> worker) {
+    this(worker, LWTPoolImpl.defaultPriority);
+  }
+
+  @Override
+  public boolean add(T e) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public boolean contains(Object o) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public int drainTo(Collection<? super T> c, int maxElements) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public int drainTo(Collection<? super T> c) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public boolean offer(T e, long timeout, TimeUnit unit) throws InterruptedException {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public boolean offer(T e) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public void put(T e) throws InterruptedException {
+    if (Log.levelMet(Level.FINE)) {
+      Log.fine("LWTQueueImpersonator.put ", e);
+    }
+    worker.addPrioritizedWork(e, priority);
+  }
+
+  public void putAll(Collection<? extends T> c) throws InterruptedException {
+    for (T t : c) {
+      put(t);
+    }
+  }
+
+  @Override
+  public T poll(long timeout, TimeUnit unit) throws InterruptedException {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public T take() throws InterruptedException {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public int remainingCapacity() {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public boolean remove(Object o) {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public T element() {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public T peek() {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public T poll() {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public T remove() {
+    throw new UnsupportedOperationException("Method not implemented");
+  }
+
+  @Override
+  public boolean addAll(Collection<? extends T> c) {
+    try {
+      putAll(c);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)